pkg/pgclient/config.go

```diff
@@ -154,32 +158,76 @@ func (cfg *Config) GetConnectionStr() string {
 	return cfg.DbUri
 }
 
-// GetPoolSize returns the max pool size based on the max_connections allowed by database and the defaultFraction.
-// Arg inputPoolSize is the pool size provided in the CLI flag.
-func (cfg *Config) GetPoolSize(poolName string, defaultFraction float64, inputPoolSize int) (int, error) {
-	maxConns, err := cfg.maxConn()
-	if err != nil {
-		return 0, fmt.Errorf("max connections: %w", err)
+// GetPoolSizes returns the pool sizes adjusted according to defaults, read only settings, and max connections available
+func (cfg *Config) GetPoolSizes(readOnly bool, dbMaxConns int) (readerPoolSize, writerPoolSize, maintPoolSize int, err error) {
```

Comment:
We could split this function in 2, getting the sizes and validating them. Maybe something like:

```
GetPoolSizes {
   poolSizes := doGetPoolSizes()
   err := validatePoolSizes(poolSizes)
}
```

WDYT?

Reply:
I'm splitting out the validation. I think once I do that and the other suggestions, the function is small enough to leave the calculation of the sizes inline.